levels/lovelace_tower/events.js

```diff
@@ -7,347 +7,78 @@ const INITIAL_STATE = {
-    );
+if (event.name === "triggerAreaWasEntered" && event.target.key === "objectiveBarrier2" && worldState.obj2Complete === false){
+  world.showNotification(
+    "I need to complete the next objective before I can pass through to the next area."
+  );
+};
 
-    // TODO: actually add item to inventory
-    tweenToScroll();
-  };
+if (event.name === "triggerAreaWasEntered" && event.target.key === "objectiveBarrier3" && worldState.obj3Complete === false){
+  world.showNotification(
+    "I need to complete the next objective before I can pass through to the next area."
+  );
+};
 
-  const addPledgeScroll = (event) => {
-    worldState.insideCatacombs.hasPledgeScroll = true;
-    destroyObject("pledge_scroll");
-    world.showNotification(
-      "I've obtained my pledge scroll. Time to head to the Academy building and present it to the headmaster."
-    );
+// Remove objective barriers
+if (worldState.obj1Complete === true){
+  world.hideEntities(`api-hidden-door-1`);
+};
+if (worldState.obj2Complete === true){
+  world.hideEntities(`api-hidden-door-2`);
+};
+if (worldState.obj3Complete === true){
+  world.hideEntities(`api-hidden-door-3`);
+};
 
-    world.grantItems(["pledge_scroll"]);
-    determineHouse();
+// Once the final objective has been hacked and closed, hide books and empty shelves
+if (worldState.obj5Complete === true){
+  if (event.name === "objectiveDidClose" && event.target.objectiveName === "api-05-get-patch"){
+    world.hideEntities(`bad-book`);
   };
+  
+  world.enableTransitionAreas("exit_to_library_corridor");
 
-  // OTHER???
-
-  const runAddItem = (event) => {
-    console.log(event);
-    items[event.target.key](event);
+  // As player tries to leave, trigger Fredric conversation
+  if (event.name === "triggerAreaWasEntered" && event.target.key === "fredricTrigger" && worldState.fredricNoteTriggered === false){
+    world.startConversation("fredric-threat-lovelace", "fredricNeutral.png");
   };
 
-  const tweenToScroll = () => {
-    if (worldState.insideCatacombs.tweenRunning) {
-      return;
-    }
-
-    worldState.insideCatacombs.tweenRunning = true;
-
-    world.forEachEntities("scroll_viewpoint", async (viewpoint) => {
-      console.log(viewpoint);
-      world.disablePlayerMovement();
-
-      await world.tweenCameraToPosition({
-        x: viewpoint.startX,
-        y: viewpoint.startY,
-      });
-
-      await world.wait(3000);
-      await world.tweenCameraToPlayer();
-
-      world.enablePlayerMovement();
-      worldState.insideCatacombs.tweenRunning = false;
+  // When Fredric trigger closes, Fredric letter becomes interactable 
+  if (event.name === "conversationDidEnd" && event.npc.conversation === "fredric-threat-lovelace"){
+    world.forEachEntities("fredricNote", note => {
+      note.interactable = true;
     });
+    worldState.fredricNoteTriggered = true;
+    // world.getState(HOUSE_CEREMONY_STATE_KEY) set houseLovelaceComplete = true;
```

Comment:
Yeah, sorry this was me not sure exactly how to set that variable across different levels. It won't really matter for this release because the player won't be able to do anything after completing Lovelace Tower. But eventually we will want to track that progress since the player will be continually jumping back to the house_ceremony level in order to access future houses.